### ISSUE TITLE
project pull results in error: AttributeError: 'DescriptorPullMutator' object has no attribute 'journal'

### ISSUE THREAD

**Issue by dvaccarosenna** [accanto-systems/lmctl#5]
Version: lmctl 2.1.0

Executing a `project pull` on a project with a descriptor that references a descriptor not in the same project results in an error:

```
--> Pull Sources
Pulling descriptor for test
Pulling descriptor assembly::test::1.0 from LM (https://192.168.56.100:8083)
Creating backup of descriptor ./Descriptor/assembly.yml
Saving pulled descriptor to ./Descriptor/assembly.yml
Traceback (most recent call last):
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/mutate/descriptor.py", line 57, in __convert_types
    resolved_project = self.config_references.resolve(project_for_descriptor_reference)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/source/config_references.py", line 20, in resolve
    return super().resolve(reference, self.resolution_map)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/reference.py", line 57, in resolve
    return ReferenceResolver(self.schema, resolution_map).resolve(reference)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/reference.py", line 93, in resolve
    return self.__resolve(reference, reference_parts, self.resolution_map)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/reference.py", line 105, in __resolve
    return self.__resolve(reference, reference_parts, next_value)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/reference.py", line 100, in __resolve
    raise NotResolvableError('Cannot find \'{0}\' in reference: {1}'.format(next_part, reference))
lmctl.reference.NotResolvableError: Cannot find 'resource::hello-world::1.0' in reference: $lmctl:/descriptor_mappings:/resource::hello-world::1.0:/project
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/bin/lmctl", line 10, in <module>
    sys.exit(init_cli())
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/cli/entry.py", line 23, in init_cli
    cli()
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/cli/commands/project.py", line 140, in pull
    exec_pull(controller, project, env_sessions)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/cli/commands/project.py", line 63, in exec_pull
    controller.execute(project.pull, env_sessions, pull_options)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/cli/lifecycle.py", line 252, in execute
    response = exec_func(*args)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/source/core.py", line 184, in pull
    return self.__do_pull(env_sessions, options, journal)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/source/core.py", line 188, in __do_pull
    pull_exec.PullProcess(self, options, journal, env_sessions).execute()
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/processes/pull.py", line 44, in execute
    return PullWorker(self.project, self.options, backup_tree, self.journal, self.env_sessions, self.references).work()
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/processes/pull.py", line 57, in work
    self.__pull_sources()
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/processes/pull.py", line 64, in __pull_sources
    self.project.source_handler.pull_sources(self.journal, backup_tool, self.env_sessions, self.references)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/handlers/assembly/assembly_src.py", line 241, in pull_sources
    self.__pull_descriptor(journal, backup_tool, backup_tree, lm_session, references)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/handlers/assembly/assembly_src.py", line 260, in __pull_descriptor
    descriptor = descriptor_mutations.DescriptorPullMutator(references).apply(descriptor)
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/mutate/descriptor.py", line 44, in apply
    self.__convert_types(descriptor.raw, 'composition')
  File "/mnt/c/Users/hardw/git/lmctl_2/lmctl/env/lib/python3.5/site-packages/lmctl/project/mutate/descriptor.py", line 61, in __convert_types
    self.journal.event('Cannot resolve reference: {0}'.format(element_type))
AttributeError: 'DescriptorPullMutator' object has no attribute 'journal'
```

**Comment by dvaccarosenna** [comment]
The `lmctl.reference.NotResolvableError` part of this is expected. The handler of this error is referencing `self.journal` which doesn't exist but should.

**Comment by dvaccarosenna** [comment]
Fixed by PR #6

**Comment by dvaccarosenna** [comment]
Fixed by #6
